sdk/formrecognizer/azure-ai-formrecognizer/migration-guide.md

```diff
@@ -0,0 +1,255 @@
+        }
+    }
+
+    DocumentField merchantPhoneNumberField = receiptFields.get("MerchantPhoneNumber");
+    if (merchantPhoneNumberField != null) {
+        if (DocumentFieldType.PHONE_NUMBER == merchantPhoneNumberField.getType()) {
+            String merchantAddress = merchantPhoneNumberField.getValuePhoneNumber();
+            System.out.printf("Merchant Phone number: %s, confidence: %.2f%n",
+                merchantAddress, merchantPhoneNumberField.getConfidence());
+        }
+    }
+
+    DocumentField transactionDateField = receiptFields.get("TransactionDate");
+    if (transactionDateField != null) {
+        if (DocumentFieldType.DATE == transactionDateField.getType()) {
+            LocalDate transactionDate = transactionDateField.getValueDate();
+            System.out.printf("Transaction Date: %s, confidence: %.2f%n",
+                transactionDate, transactionDateField.getConfidence());
+        }
+    }
+
+    DocumentField receiptItemsField = receiptFields.get("Items");
+    if (receiptItemsField != null) {
+        System.out.printf("Receipt Items: %n");
+        if (DocumentFieldType.LIST == receiptItemsField.getType()) {
+            List<DocumentField> receiptItems = receiptItemsField.getValueList();
+            receiptItems.stream()
+                .filter(receiptItem -> DocumentFieldType.MAP == receiptItem.getType())
+                .map(formField -> formField.getValueMap())
+                .forEach(formFieldMap -> formFieldMap.forEach((key, formField) -> {
+                    if ("Name".equals(key)) {
+                        if (DocumentFieldType.STRING == formField.getType()) {
+                            String name = formField.getValueString();
+                            System.out.printf("Name: %s, confidence: %.2fs%n",
+                                name, formField.getConfidence());
+                        }
+                    }
+                    if ("Quantity".equals(key)) {
+                        if (DocumentFieldType.FLOAT == formField.getType()) {
+                            Float quantity = formField.getValueFloat();
+                            System.out.printf("Quantity: %f, confidence: %.2f%n",
+                                quantity, formField.getConfidence());
+                        }
+                    }
+                }));
+        }
+    }
+}
+```
+
+#### Build a custom document analysis model
+In 3.X, creating a custom model required specifying `useTrainingLabels` to indicate if the model creation was using the custom labeling or not with the `beginTraining` method.
+In 4.X, we introduced the new general document model (prebuilt-document) consisting of key-value pair extraction, it is no longer necessary to train a custom model to extract key-value pairs. Thus, we removed the option to train custom models without labels. Hence, with 4.X, we can build a custom document analysis model using the `beginBuildModel` method as shown in the example below:
```

Comment:
we never exposed key-value pairs in v3 - I would instead say something about the "train without labels" scenario